build.xml

```diff
@@ -221,7 +214,15 @@
       <arg value="--always"/>
     </exec>
     <!-- Set version if git describe failed. -->
-    <property name="adaptorlib.version" value="unknown"/>
+    <property name="adaptorlib.version" value="unknown-${DSTAMP}"/>
+
+    <!-- Use the version without the "v" prefix for file names. -->
+    <loadresource property="adaptorlib.suffix">
+      <propertyresource name="adaptorlib.version"/>
+      <filterchain>
+          <replaceregex pattern="^(v(\d))?" replace="-\2"/>
```

Comment:
want to double check:

everything is going to match pattern on left because everything has a beginning and the rest of pattern is optional.

if version starts with "v321" then suffix will become "-321".

if version="unknown=2017" then suffix becomes .....  "-"? I'm not sure if I read that right, I'm guessing since 2nd grouping didn't match anything then we just get the "-".  Is that right reading?